cpp/src/utilities/host_memory.cpp

```diff
@@ -277,7 +279,7 @@ bool config_default_pinned_memory_resource(pinned_mr_options const& opts)
 CUDF_EXPORT auto& kernel_pinned_copy_threshold()
 {
   // use cudaMemcpyAsync for all pinned copies
-  static std::atomic<size_t> threshold = 0;
+  static std::atomic<size_t> threshold = getenv_or("LIBCUDF_KERNEL_PINNED_COPY_THRESHOLD", 0);
```

Comment:
One issue with static variable is that, they are initialized only once. The user will not be able to change them after they are set. Can we remove static?

Reply:
I'm torn on this one. You're right, but:
I really like reading the env var once so I can log it cleanly (already a part of `getenv_or`).
Also, there are (C++) APIs to set these thresholds at runtime.
I'd keep this static until we have a use case that requires runtime changes.